File: backend/src/main/java/com/form/org/Services/FournisseurService.java
package com.form.org.Services;

import com.form.org.dto.FournisseurDTO;

import java.util.Date;
import java.util.List;

public interface FournisseurService {
    FournisseurDTO save(FournisseurDTO dto);

    FournisseurDTO findById(Integer idFournisseur);

    FournisseurDTO findByNom(String nom);

    List<FournisseurDTO> findAll();

    void delete(Integer idFournisseur);
}

File: FrontEnd1/src/app/services/fournisseur/fournisseur.service.ts
import { Injectable } from '@angular/core';
import { Observable, of } from 'rxjs';
import { FournisseurDTO } from 'src/gs-api/src/models/fournisseur-dto';
import { FournisseurService } from '../../../gs-api/src/services/fournisseur.service';

@Injectable({
  providedIn: 'root'
})
export class FournisseurServices {

  constructor(
    private fournisseurService : FournisseurService
    ) { };

   enregistrerfournisseur(fournisseurDt:FournisseurDTO):Observable<FournisseurDTO> {
     return this.fournisseurService.save(fournisseurDt);
   }
   findAllffournisseur(): Observable<FournisseurDTO[]> {
    return this.fournisseurService.findAll();
  }

  findFournisseurById(idFournisseur?: number): Observable<FournisseurDTO> {
    if (idFournisseur) {
      return this.fournisseurService.findById(idFournisseur);
    }
    return of();
  }
  deleteFournisseur(idFournisseur?: number): Observable<any>{
    if (idFournisseur) {
      return this.fournisseurService.delete(idFournisseur);
    }
    return of();
  }
 

}
